packages/aws-cdk-lib/aws-route53/lib/record-set.ts

```diff
@@ -929,6 +929,10 @@ export class CrossAccountZoneDelegationRecord extends Construct {
       throw Error('Only one of parentHostedZoneName and parentHostedZoneId is supported');
     }
 
+    if (!props.delegatedZone.hostedZoneNameServers) {
```

Comment:
nit: I think it'd be nice to have some read me changes [here](https://github.com/aws/aws-cdk/blob/ddbbd002da6298679500d4dd6b6c5b1487bd5f5d/packages/aws-cdk-lib/aws-route53/README.md?plain=1#L246)

Reply:
Thanks for the callout! I updated the README with a new commit.